### PR TITLE
FIX - sn2individual: "File too small"

### PR DESCRIPTION
**Pull request by tiborauer**
This PR resolves "File too small" error by reloading the saved deformation before resampling.

### CODE REVIEW

File: utilities/private/sn2individual.m [hunk 1]
```diff
@@ -86,8 +86,7 @@
   V.descrip = 'dummy volume';
   create(V);
   V.dat(:) = 0;
-  P.image(1).private = V;
-  P.image(1).fname   = fname;
+  P.image(1) = spm_vol(fname);
```
Comment (schoffelen) [inline utilities/private/sn2individual.m:89]: Can you provide a test function (or add your faulty use case to an existing test function) to demonstrate the error in your particular use case, and to ensure that other (hencetoforth) non-erroneous use cases still work with the suggested changes in place? Thanks.

**Review by schoffelen — CHANGES_REQUESTED** [review]

File: utilities/private/sn2individual.m (hunk 1 above)
Comment (robertoostenveld) [reply utilities/private/sn2individual.m:89]: I specifically wonder if this is SPM version specific, with "SPM version" as in a subtle difference that FT is not aware of or sensitive to. Like from SPM12.1 to SPM12.2, if such versions would exist.

**Comment by tiborauer** [comment]
Here is a minimal working example, where the original function fails:

[Here](https://www.dropbox.com/sh/3bpyq7mmsxxspgr/AABD9eH3kOWSNJr7Wu5VEsgla?dl=0) you can find the relevant files. You may need to change the path to the standard sourcemodel.

```
mri = ft_read_mri('mri.nii','dataformat','nifti_spm');
mri.coordsys = 'ras';
cfg     = [];
cfg.dim = mri.dim;
mri     = ft_volumereslice(cfg,mri);

cfg = [];
cfg.spmversion = 'spm12';
cfg.spmmethod = 'new';
cfg.nonlinear = 'no';
normalise = ft_volumenormalise(cfg, mri);

mnigrid = load('/users/psychology01/software/fieldtrip/template/sourcemodel/standard_sourcemodel3d10mm.mat');
mnigrid = mnigrid.sourcemodel;

sourcemodel.pos = ft_warp_apply(inv(normalise.initial), ft_warp_apply(normalise.params, mnigrid.pos, 'sn2individual'));
```

The issue may not be the `sn2individual` because when I tried the script with standard SPM segmentation+normalisation structure (see seg8.mat in the shared folder), it worked:

```
seg = load('seg8.mat');
sn2individual(seg,mnigrid.pos)
```

**Comment by schoffelen** [comment]
Hi Tibor, 
The idea would be that you write a test function, and put it in fieldtrip/test
You can call the file test_issue1695, and copy the necessary overhead stuff (i.e. the first lines) of any of the other test functions in the folder.

**Comment by schoffelen** [comment]
Good point. Typically we rely on a file that is located (on a known location) on our fileserver.  If the test data is really specific to the test, and small in size we typically add it (also locally to a dedicated test folder, which you don't have access to). 
I think that for this purpose the exact anatomical does not matter. What I do sometimes (in order to be able to run a test function that relies on a file that is otherwise located behind the firewall) is to have an optional input argument (e.g. the fullpathandfilename to the file), and to have an if-end to handle the case of no input arguments (which then will switch to load a file from a hard coded location).

**Comment by tiborauer** [comment]
I have added the test. I use the signle_subject anatomy in FieldTrip, so no external file is required.

**Review by schoffelen — CHANGES_REQUESTED** [review]

File: test/test_issue1695.m [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+cfg.spmversion = 'spm12';
+cfg.spmmethod = 'new';
+cfg.nonlinear = 'no';
+normalise = ft_volumenormalise(cfg, mri);
+
+% this will likely crash
+pos = ft_warp_apply(normalise.params, sourcemodel.pos, 'sn2individual');
+
+%% Normalise #2 - SPM
+% Full version of SPM12 is required
```
[the remark this answers is not in the document]
Comment (tiborauer) [reply test/test_issue1695.m:25]: FieldTrips's spm12 toolbox has not all the functionality. In particular, it does not have `spm_preproc_run`. The whole point of the test was to show that the issue is likely not with the `sn2individual` but the `ft_volumenormalise` (or how it is called).

**Comment by tiborauer** [comment]
I wrote a minimal working example. It is largely based on the grid-based source model tutorial, and the explicit calls mimics https://github.com/fieldtrip/fieldtrip/blob/5efe78b3fee17f9c3d24291ffc325e49f072f943/ft_prepare_sourcemodel.m#L651-L668

File: test/test_issue1695.m [hunk 3]
```diff
@@ -0,0 +1,54 @@
+%% Init
```
[the remark this answers is not in the document]
Comment (tiborauer) [reply test/test_issue1695.m:7]: Done.

File: test/test_issue1695.m [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+cfg.nonlinear = 'no';
+normalise = ft_volumenormalise(cfg, mri);
+
+% this will likely crash
+pos = ft_warp_apply(normalise.params, sourcemodel.pos, 'sn2individual');
+
+%% Normalise #2 - SPM
+% Full version of SPM12 is required
+
+tmpfile = spm_file(tempname,'ext','nii');
```
[the remark this answers is not in the document]
Comment (tiborauer) [reply test/test_issue1695.m:33]: Yes, I tried that first, but as I said above, it lacks an SPM functions my example is using. Do you suggest extending the SPM12 toolbox?

**Comment by schoffelen** [comment]
My bad, I just screwed up this branch. Stay tuned, I will try and fix this, don't do anything to the code in the mean time

**Comment by schoffelen** [comment]
Could you do a force push to origin of this branch, assuming that you haven't pulled in the meantime, this would clean it up:

git push -f origin norm

**Comment by schoffelen** [comment]
thanks

**Comment by schoffelen** [comment]
As discussed in #1700, this one will be closed without merging.

**Pull request closed without merging**